servant-foreign/src/Servant/Foreign/Internal.hs

```diff
@@ -9,6 +9,7 @@ module Servant.Foreign.Internal where
 
 import           Control.Lens (makePrisms, makeLenses, Getter, (&), (<>~), (%~),
                                (.~))
+import           Data.Data
```

Comment:
seems the unrestricted import breaks `-Werror` as `Data.Data` exports `Proxy` too. Could you use explicit import lists, maybe then travis will be happy?